diff --git a/tools/qemu-xen/include/net/net.h b/tools/qemu-xen/include/net/net.h
index 5d150808..1de01433 100644
--- a/tools/qemu-xen/include/net/net.h
+++ b/tools/qemu-xen/include/net/net.h
@@ -221,7 +221,7 @@ NetClientState *net_hub_port_find(int hub_id);
 
 #define DEFAULT_NETWORK_SCRIPT CONFIG_SYSCONFDIR "/qemu-ifup"
 #define DEFAULT_NETWORK_DOWN_SCRIPT CONFIG_SYSCONFDIR "/qemu-ifdown"
-#define DEFAULT_BRIDGE_HELPER CONFIG_QEMU_HELPERDIR "/qemu-bridge-helper"
+#define DEFAULT_BRIDGE_HELPER CONFIG_QEMU_HELPERDIR "/xen-bridge-helper"
 #define DEFAULT_BRIDGE_INTERFACE "br0"
 
 void qdev_set_nic_properties(DeviceState *dev, NICInfo *nd);
diff --git a/tools/qemu-xen/meson.build b/tools/qemu-xen/meson.build
index b3e7ec0e..9cd36a43 100644
--- a/tools/qemu-xen/meson.build
+++ b/tools/qemu-xen/meson.build
@@ -2759,7 +2759,7 @@ if have_tools
   endif
 
   if targetos == 'linux'
-    executable('qemu-bridge-helper', files('qemu-bridge-helper.c'),
+    executable('xen-bridge-helper', files('xen-bridge-helper.c'),
                dependencies: [qemuutil, libcap_ng],
                install: true,
                install_dir: get_option('libexecdir'))
diff --git a/tools/qemu-xen/qemu-bridge-helper.c b/tools/qemu-xen/qemu-bridge-helper.c
index a26e1663..8abc0e74 100644
--- a/tools/qemu-xen/qemu-bridge-helper.c
+++ b/tools/qemu-xen/qemu-bridge-helper.c
@@ -68,7 +68,7 @@ typedef QSIMPLEQ_HEAD(ACLList, ACLRule) ACLList;
 static void usage(void)
 {
     fprintf(stderr,
-            "Usage: qemu-bridge-helper [--use-vnet] --br=bridge --fd=unixfd\n");
+            "Usage: xen-bridge-helper [--use-vnet] --br=bridge --fd=unixfd\n");
 }
 
 static int parse_acl_file(const char *filename, ACLList *acl_list)
diff --git a/tools/qemu-xen/qemu-options.hx b/tools/qemu-xen/qemu-options.hx
index 83aa59a9..87a898bd 100644
--- a/tools/qemu-xen/qemu-options.hx
+++ b/tools/qemu-xen/qemu-options.hx
@@ -2910,7 +2910,7 @@ SRST
     If running QEMU as an unprivileged user, use the network helper
     to configure the TAP interface and attach it to the bridge.
     The default network helper executable is
-    ``/path/to/qemu-bridge-helper`` and the default bridge device is
+    ``/path/to/xen-bridge-helper`` and the default bridge device is
     ``br0``.
 
     ``fd``\ =h can be used to specify the handle of an already opened
@@ -2936,14 +2936,14 @@ SRST
         #launch a QEMU instance with the default network helper to
         #connect a TAP device to bridge br0
         |qemu_system| linux.img -device virtio-net-pci,netdev=n1 \\
-                -netdev tap,id=n1,"helper=/path/to/qemu-bridge-helper"
+                -netdev tap,id=n1,"helper=/path/to/xen-bridge-helper"
 
 ``-netdev bridge,id=id[,br=bridge][,helper=helper]``
     Connect a host TAP network interface to a host bridge device.
 
     Use the network helper helper to configure the TAP interface and
     attach it to the bridge. The default network helper executable is
-    ``/path/to/qemu-bridge-helper`` and the default bridge device is
+    ``/path/to/xen-bridge-helper`` and the default bridge device is
     ``br0``.
 
     Examples:
